### PR TITLE
CI pipeline changes 

### PR DESCRIPTION
**Pull request by sameer2209-arm**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                  }
+                  else{
+                    sh "go get -u github.com/armPelionEdge/devicedb && cd /home/jenkins/goprojects/src/github.com/armPelionEdge/devicedb && git fetch --all && git checkout ${env.BRANCH_NAME} && git merge origin/${env.BRANCH_NAME} && go build"
+                  }
+              }
+          }
+       }
+    }
+    
+    //stage('Test and Code Review') {
```
Comment (cmonr) [inline Jenkinsfile]: Please remove and fix indentation, or enable.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      //parallel {
+        stage('Test'){
+          agent{
+            label 'noi-linux-ubuntu16-ci-slave'
+          }
+          steps {
+            catchError(buildResult: 'SUCCESS'){
```
Comment (cmonr) [inline Jenkinsfile]: What's the reason that this is here?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        }
+        options{
+            lock resource: 'noi-linux-ubuntu16-ci-slave'
+        }
+        steps {
+         //checkout scm
+          withEnv(["GOROOT=/home/jenkins/go", "GOPATH=/home/jenkins/goprojects", "PATH+GO=/home/jenkins/goprojects/bin:/home/jenkins/go/bin"]){
+              script{
+                  sh 'rm -rf /home/jenknis/goprojects/src/github.com/armPelionEdge/devicedb'
+                  if(env.BRANCH_NAME ==~ /^PR-[0-9]*$/){
```
[the remark this answers is not in the document]
Comment (sameer2209-arm) [reply Jenkinsfile]: @cmonr Yes, in a multibranch pipeline, for our configurations a PR branch always starts with a PR followed by the PR number.

File: Jenkinsfile (hunk 1 above)
Comment (sameer2209-arm) [reply Jenkinsfile]: okay will do that

File: Jenkinsfile (hunk 2 above)
Comment (sameer2209-arm) [reply Jenkinsfile]: As far as I remember that was required for devicejs but while I was copying certain things I missed removing this part.

File: Jenkinsfile (hunk 1 above)
Comment (sameer2209-arm) [reply Jenkinsfile]: Fixed the indentation

File: Jenkinsfile (hunk 2 above)
Comment (sameer2209-arm) [reply Jenkinsfile]: Removed that line. It was not necessary in this case.

**Comment by sameer2209-arm** [comment]
Please select **squash and merge** option while merging.

**Pull request merged**